plugins/node/opentelemetry-instrumentation-fastify/test/instrumentation.test.ts

```diff
@@ -338,12 +338,16 @@ describe('fastify', () => {
 
           // done was not yet called from the hook, so it should not end the span
           const preDoneSpans = getSpans().filter(
-            s => !s.attributes[AttributeNames.PLUGIN_NAME]
+            s =>
+              !s.attributes[AttributeNames.PLUGIN_NAME] ||
+              s.attributes[AttributeNames.PLUGIN_NAME] === 'fastify'
```

Comment:
Can you explain what changed so that now we need this extra check?

Reply:
Absolutely. Fastify reworked their namespacing of plugins. All plugins are now under the fastify top level namespace.

What used to be `"plugin.name": "my-plugin"` is now `"plugin.name": "fastify -> my-plugin"`.

The extra check is there so that if you wanted to switch between fastify versions and run tests, you could and still have consistent results.

Fastify 3.x checking if the plugin name does not exist is enough, for Fastify 4.x the equivalent is to check if the plugin name is just `fastify`.

If you would like me to remove it from the tests as you don't think y'all will be switching versions, I am totally happy to.